pilot/pkg/xds/ads.go

```diff
@@ -180,13 +186,28 @@ func (s *DiscoveryServer) processRequest(req *discovery.DiscoveryRequest, con *C
 		s.StatusReporter.RegisterEvent(con.ConID, req.TypeUrl, req.ResponseNonce)
 	}
 
-	if !s.shouldRespond(con, req) {
+	shouldRespond := s.shouldRespond(con, req)
+
+	con.proxy.Lock()
+	request, haveBlockedPush := con.blockedPushes[req.TypeUrl]
+	delete(con.blockedPushes, req.TypeUrl)
+	con.proxy.Unlock()
+
+	if shouldRespond {
```

Comment:
Would it help to simplify this if condition if `shouldRespond` returns the RequestType also (like ACK, NACK etc.. as shouldRespond already determines that)? Not a big deal but just throwing it out

Reply:
I wrote this initially but its actually not just ACK/NACK/respond, there is "init", "reconnect", etc. If we treat all of those as "ack" then it may be simpler